.gitlab/nersc-gitlab-ci.yml

```diff
@@ -0,0 +1,82 @@
+# Kokkos nightly tests at NERSC machine Perlmutter.
+stages:
+  - test
+  - clean_up
+
+default:
+  tags:
+    - perlmutter-login
+  before_script:
+    - export SOURCE_DIR=$SCRATCH/.nightly/testing/kokkos
+    - if [ ! -d $SOURCE_DIR ]; then
+    - git clone https://github.com/kokkos/kokkos.git ${SOURCE_DIR}
+    - fi
+    - cd ${SOURCE_DIR} && pwd && git pull && git branch
+    - export CMAKE_BUILD_PARALLEL_LEVEL=32
+    - export ENV_CMAKE_OPTIONS=""
+    - export ENV_CMAKE_OPTIONS="${ENV_CMAKE_OPTIONS};-DKokkos_ENABLE_TESTS=ON"
+    - export SCRIPT_FILE=${SOURCE_DIR}/scripts/CTestRun.cmake
+    - export OMP_NUM_THREADS=4
+    - export OMP_PROC_BIND=spread
+    - export OMP_PLACES=cores
+
+EPYC-OMP-gcc:
+  stage: test
+  script:
+    - echo "Kokkos-OpenMP tests by $GITLAB_USER_LOGIN using gcc."
+    - module load PrgEnv-gnu gcc/12.2.0
+    - export BUILD_DIR=${SOURCE_DIR}/build_omp_gcc
+    - export ENV_CMAKE_OPTIONS="${ENV_CMAKE_OPTIONS};-DCMAKE_CXX_COMPILER=g++"
+    - export ENV_CMAKE_OPTIONS="${ENV_CMAKE_OPTIONS};-DKokkos_ARCH_ZEN3=ON"
+    - export ENV_CMAKE_OPTIONS="${ENV_CMAKE_OPTIONS};-DKokkos_ENABLE_OPENMP=ON"
```

Comment:
Can we enable compiler warnings and add `-Werror` to all the builds?

Reply:
Yes, that's we are doing in most other CI. At least, we are able to see warnings when reporting to `CDash` (as opposed to other systems where warnings would just get ignored).

Reply:
No we don't want -Werror on nightly builds